homeassistant/components/knx/switch.py

```diff
@@ -34,32 +36,33 @@ async def async_setup_entry(
     async_add_entities: AddEntitiesCallback,
 ) -> None:
     """Set up switch(es) for KNX platform."""
-    xknx: XKNX = hass.data[DOMAIN].xknx
-    config: list[ConfigType] = hass.data[DATA_KNX_CONFIG][Platform.SWITCH]
+    knx_module: KNXModule = hass.data[DOMAIN]
 
-    async_add_entities(KNXSwitch(xknx, entity_config) for entity_config in config)
+    yaml_config: list[ConfigType] | None
+    if yaml_config := hass.data[DATA_KNX_CONFIG].get(Platform.SWITCH):
```

Comment:
Side note: do we have multiple places within `hass.data` where we store KNX data? Ideally we only use the `hass.data[DOMAIN]`

Reply:
Yeah, there is the yaml config in `DATA_KNX_CONFIG` and an instance of the base class `KNXModule` in `DOMAIN`. I recall there was a reason why we did it that way, but tbh I can't remember why anymore 😬 had something to do with config entry reloads 🤔 
This should be addressed together with typing improvements of hass.data I guess.